### PR TITLE
Predictive targeting

### LINKED ISSUE #84: Develop Predictive targeting

### PR DESCRIPTION
**Pull request by Sirkular**
Ranged and turret enemies will now spawn with predictive targeting 50% of the time.
Bullet speeds for ranged and turret is defined in a dictionary in Globals. Turret bullet speed has been bumped from 10 -> 20 so predictive targeting math doesn't go undefined.
image

closes #84

### CODE REVIEW

File: Assets/Scripts/Enemy/EnemySpawn.cs [hunk 1]
```diff
@@ -194,11 +194,16 @@ void SpawnEnemy(EnemyType enemy)
...
         }
 
-        random = Random.Range(0, 10);
-        if (enemyType == EnemyType.TURRET && random < 5)
+        // 50% to be predictive if turret or ranged.
+        if ((enemyType == EnemyType.TURRET || enemyType == EnemyType.RANGED) && Random.Range(0, 2) == 0)
+        {
+            variant = EnemyVariantType.PREDICTIVE;
+        }
+        else if (enemyType == EnemyType.TURRET && Random.Range(0, 2) == 0)
```
Comment (zhuandy42) [inline Assets/Scripts/Enemy/EnemySpawn.cs:202]: max is gonna scream

**Review by zhuandy42 — APPROVED** [review]
pog

File: Assets/Scripts/Enemy/EnemySpawn.cs (hunk 1 above)
Comment (Sirkular) [reply Assets/Scripts/Enemy/EnemySpawn.cs:202]: Got a shiny new ticket for this haha:
https://github.com/MaXeraph/Adrenarun/issues/93

Comment (MaXeraph) [reply Assets/Scripts/Enemy/EnemySpawn.cs:202]: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa

**Review by MaXeraph — APPROVED** [review]

**Pull request merged**